### PR TITLE
adding make local-docker-image

### LINKED ISSUE #12644: build: Local arm64 images
### Proposal

The instructions to build docker images locally is the following:

```
make promu
promu crossbuild -p linux/amd64
make npm_licenses
make common-docker-amd64
```

However, this only builds an amd64 image even on an ARM system (like the newer Macs). We should a make target that builds native architecture images.

### PR DESCRIPTION
**Pull request by sbdtu5498**
Fixes #12644 
cc: @gouthamve

### CODE REVIEW

**Review by roidelapluie — COMMENTED** [review]
Can we somehow just build the docker image for the current arch ?

**Comment by gouthamve** [comment]
Can you elaborate further @roidelapluie? This does build the image for the current arch if I understood correctly.

File: Makefile [hunk 1]
```diff
@@ -112,6 +114,16 @@ tarball: npm_licenses common-tarball
 .PHONY: docker
 docker: npm_licenses common-docker
 
+.PHONY: local-docker-image $(BUILD_LOCAL_DOCKER_ARCHS)
+local-docker-image: $(BUILD_LOCAL_DOCKER_ARCHS)
+$(BUILD_LOCAL_DOCKER_ARCHS): local-docker-image-%: 
+	promu crossbuild-natively common-docker-$* npm_licenses 
+
+.PHONY: crossbuild-natively
+crossbuild-natively:
```
Comment (bwplotka) [inline Makefile:123]: Hm, it's a bit inconsistent e.g. we don't have `crossbuild` Makefile cmd. Can we remove it?

File: Makefile [hunk 2]
```diff
@@ -112,6 +114,16 @@ tarball: npm_licenses common-tarball
 .PHONY: docker
 docker: npm_licenses common-docker
 
+.PHONY: local-docker-image $(BUILD_LOCAL_DOCKER_ARCHS)
+local-docker-image: $(BUILD_LOCAL_DOCKER_ARCHS)
```
Comment (bwplotka) [inline Makefile:118]:
```suggestion
docker-current-arch: $(BUILD_LOCAL_DOCKER_ARCHS)
```

File: Makefile [hunk 3]
```diff
@@ -23,6 +23,8 @@ TSDB_BENCHMARK_NUM_METRICS ?= 1000
 TSDB_BENCHMARK_DATASET ?= ./tsdb/testdata/20kseries.json
 TSDB_BENCHMARK_OUTPUT_DIR ?= ./benchout
 
+BUILD_LOCAL_DOCKER_ARCHS = $(addprefix local-docker-image-,$(DOCKER_ARCHS))
```
Comment (bwplotka) [inline Makefile:26]: Hm, that still has all architectures, not only local one?

Comment (bwplotka) [reply Makefile:26]: What's `local-docker-image-` is this something that promu understands?

**Review by bwplotka — COMMENTED** [review]
Nice! 

Sounds like we would love to have `docker-curr-arch` or something that will build only one image with same architecture as your PC/laptop, right?

Left some comments, if we agree on the goal here (: 

(During bug scrub)

**Comment by machine424** [comment]
How about

```
.PHONY: docker-current-arch
docker-current-arch: promu
       $(PROMU) crossbuild -p linux/$(GOHOSTARCH)
       $(MAKE) npm_licenses
       $(MAKE) common-docker-$(GOHOSTARCH)
```

I didn't test that but I think it should run the needed commands.
I'm not even sure if it's optimal to run`promu crossbuild` on an ARM machine to produce an ARM build. I think amd images would be used...

**Comment by bboreham** [comment]
Hello from the bug-scrub!  Closing since no response from @sbdtu5498; @kartikaysaxena's PR is #13511.

**Pull request closed without merging**